src/System.Net.Http/src/System/Net/Http/Headers/HeaderDescriptor.cs

```diff
@@ -111,7 +114,42 @@ public string GetHeaderValue(ReadOnlySpan<byte> headerValue)
                 }
             }
 
+            if (KnownHeader != null && KnownHeader.Name == KnownHeaders.Location.Name)
```

Comment:
> KnownHeader.Name == KnownHeaders.Location.Nam [](start = 38, length = 46)

Could you confirm if this comparison is ok to have it as case sensitive?

Reply:
Seems it is case insensitive but it is already normalized:
https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/src/System/Net/Http/Headers/KnownHeaders.cs#L381
so check should be fine as is

Reply:
Why do we need to compare the Name... why can't we just compare the KnownHeader instance itself, e.g. `if (KnownHeader == KnownHeaders.Location)`?

Reply:
Also, a) why are we checking KnownHeader here when the previous if checks _knownHeader, and b) why can't we do this check as part of the previous check on _knownHeader != null? e.g.
```C#
if (_knownHeader != null)
{
    if (_knownHeader == KnownHeaders.Location)
    {
        ...
    }
    else
    {
        string[] knownValues = _knownHeader.KnownValues;
        if (knownValues != null)
        {
            ...
        }
    }
}
```

Reply:
@stephentoub will change - I wasn't sure we were reusing the same instance when writing that but considering my previous comment both should be equally fast (minus one comparison)